src/orion/algo/gridsearch.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.algo.gridsearch` -- Grid Search
+===========================================
+
+.. module:: gridsearch
+    :platform: Unix
+    :synopsis: Grid Search Algorithm
+
+"""
+import itertools
+import logging
+
+import numpy
+
+from orion.algo.base import BaseAlgorithm
+from orion.algo.space import Categorical, Fidelity, Integer, Real
+
+log = logging.getLogger(__name__)
+
+
+def grid(dim, num):
+    """Build a one-dim grid of num points"""
+
+    if dim.type == "categorical":
+        return categorical_grid(dim, num)
+    elif dim.type == "integer":
+        return discrete_grid(dim, num)
+    elif dim.type == "real":
+        return real_grid(dim, num)
+    elif dim.type == "fidelity":
+        return fidelity_grid(dim, num)
+    else:
+        raise TypeError(
+            "Grid Search only supports `loguniform`, `uniform` and `choices`: "
+            "`{}`".format(dim.prior_name)
+        )
+
+
+def fidelity_grid(dim, num):
+    """Build fidelity grid, that is, only top value"""
+    return [dim.interval()[1]]
+
+
+def categorical_grid(dim, num):
+    """Build categorical grid, that is, all categories"""
+    if len(dim.categories) != num:
+        log.warning(
+            f"Categorical dimension {dim.name} does not have {num} choices: {dim.categories}. "
+            "Will use {len(dim.categories)} choices instead."
+        )
+    return dim.categories
+
+
+def discrete_grid(dim, num):
+    """Build discretized real grid"""
+    grid = real_grid(dim, num)
+
+    _, b = dim.interval()
+
+    discrete_grid = [int(numpy.round(grid[0]))]
+    for v in grid[1:]:
+        int_v = int(numpy.round(v))
+        if int_v <= discrete_grid[-1]:
+            int_v = discrete_grid[-1] + 1
+
+        if int_v > b:
+            log.warning(
+                f"Cannot list {num} discrete values for {dim.name}. "
+                "Will use {len(discrete_grid)} points instead."
+            )
+            break
+
+        discrete_grid.append(int_v)
+
+    return discrete_grid
+
+
+def real_grid(dim, num):
+    """Build real grid"""
+    if dim.prior_name.endswith("reciprocal"):
+        a, b = dim.interval()
+        return list(_log_grid(a, b, num))
+    elif dim.prior_name.endswith("uniform"):
+        a, b = dim.interval()
+        return list(_lin_grid(a, b, num))
+    else:
+        raise TypeError(
+            "Grid Search only supports `loguniform`, `uniform` and `choices`: "
+            "`{}`".format(dim.prior_name)
```

Comment:
Seems same error message as the one inside the `def grid(dim, num):`

Reply:
Oh indeed, I should change the error message for grid(). Thanks!